### DISCUSSION TITLE
Projects: Allow workflows to assign an iteration

### DISCUSSION THREAD

**Discussion by talldan** [community/community#35038] [Projects and Issues]
It'd be really useful to be able to assign an iteration using a workflow.

For example, if I had an iteration called "Completed". I'd be able to set up a workflow as follows:

```
When - pr is:merged
Set - Completed: date[current]
```

This would find the iteration that 'current-date' falls into and assign it automatically.

For me, this would be useful for reporting which tasks were completed within a specific period. (There's also this suggestion as another way to achieve this - https://github.com/community/community/discussions/17085).

Another use case for this feature would be automatically assigning tasks to sprints (for example as part of this proposed feature - https://github.com/github/roadmap/issues/576):
```
When - issue Status:"Next Up"
Set - Sprint: date[current]

When - issue Status:Backlog
Set - Sprint: unset
```

That way users won't have to change both 'status' and 'sprint', the sprint is automatically handled.

**Comment by andrewheckford** [comment]
This will be useful for us too. When a PR is created from an issue we're working on in our current sprint, we'd like to auto-assign the PR to the current sprint so it can be reviewed.
